dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Simplified/VRNordic.mo

```diff
@@ -0,0 +1,166 @@
+  Modelica.Blocks.Math.Feedback dIf annotation(
+    Placement(visible = true, transformation(origin = {-240, 80}, extent = {{-10, 10}, {10, -10}}, rotation = 0)));
+  Modelica.Blocks.Math.Gain gain1(k = -1) annotation(
+    Placement(visible = true, transformation(origin = {-150, 80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Sources.Constant const2(k = IfLimPu) annotation(
+    Placement(visible = true, transformation(origin = {-270, 120}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Gain kMulDU(k = G) annotation(
+    Placement(visible = true, transformation(origin = {50, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Dynawo.Electrical.Controls.Machines.VoltageRegulators.Simplified.BaseClasses.OELNordic overExcitationLimitation(f = f, r = r, Y0 = -1) annotation(
+    Placement(visible = true, transformation(origin = {-190, 40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Gain gain(k = 10) annotation(
+    Placement(visible = true, transformation(origin = {190, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Feedback feedback1 annotation(
+    Placement(visible = true, transformation(origin = {140, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.LimIntegrator limIntegrator(outMax = EfdMaxPu, outMin = 0, y_start = Efd0Pu) annotation(
+    Placement(visible = true, transformation(origin = {230, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Feedback dOmega annotation(
+    Placement(visible = true, transformation(origin = {-240, -120}, extent = {{-10, 10}, {10, -10}}, rotation = 0)));
+  Modelica.Blocks.Sources.Constant const(k = SystemBase.omega0Pu) annotation(
+    Placement(visible = true, transformation(origin = {-270, -80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.LimIntegrator timer(outMax = 99, outMin = tOelMin, y_start = tOelMin) annotation(
+    Placement(visible = true, transformation(origin = {-150, 40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.TransferFunction leadLag(a = {tB, 1}, b = {tA, 1}, x(each fixed = true, each start = Efd0Pu), y_start = Efd0Pu) annotation(
+    Placement(visible = true, transformation(origin = {90, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Feedback dU annotation(
+    Placement(visible = true, transformation(origin = {-220, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Add add annotation(
+    Placement(visible = true, transformation(origin = {-110, -20}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  parameter Types.VoltageModulePu Efd0Pu(fixed=false) "Initial excitation voltage in pu (user-selected base voltage)";
+  parameter Types.CurrentModulePu If0Pu(fixed=false) "Initial field current in pu (base SnRef, user-selected base voltage)";
```

Comment:
I have removed the fixed=true and fixed=false, the initialization fails under OpenModelica.

Matrix singular!
Dynawo.Examples.Nordic.TestCases.TestCase_02nls.c:25258: Invalid root: (-0.116237)^(6.0257)
While solving non-linear system an assertion failed during initialization.
The non-linear solver tries to solve the problem that could take some time.
It could help to provide better start-values for the iteration variables.
For more information simulate with -lv LOG_NLS_V

If you want to get rid of these modifiers, we may have to simulate the case under Dynawo.

Reply:
Can you just keep the fixed=true then ? And get rid of the fixed=false ? That should work

Reply:
I have tried it, it did not work either.

Reply:
With the fixed=false removed, I get the same error.

Reply:
Please also change the pu base for ifPu and IfLimPu, thanks

Reply:
Base corrected for ifPu and IfLimPu.

Do you want me to change the name of the input of OELNordic from u to deltaIfPu ?